### PR TITLE
Added more input prompts and default settings

### PR DESCRIPTION
**Pull request by ana-simionescu**

### CODE REVIEW

File: user_interaction.py [hunk 1]
```diff
@@ -1,110 +1,108 @@
...
-# Determining path for data retrieval
-dataPath = raw_input("Please insert a path for data retrieving. This can be a local directory or a GCS Bucket:\n")
+# # Determining path for data retrieval
+data_path = raw_input("Insert a path for data retrieving. \
+This can be a local directory or a GCS Bucket:\n")
 
 # Determining path for storing model, snapshots and summaries
-storingPath = raw_input("\nPlease insert a path for storing the model, snapshots and summaries.\
- This can be a local directory or a GCS Bucket:\n")
+storing_path = raw_input("\nInsert a path for saving the model, \
```
Comment (werkaaa) [inline user_interaction.py]: I would go with the same variable names as in Magenta files, so I would use save_dir here.

File: user_interaction.py [hunk 2]
```diff
@@ -1,110 +1,108 @@
...
+snapshots and summaries. This can be a local directory or a GCS Bucket:\n")
 
-# Determining path for recovering a snapshot
-recoveryPath = raw_input("\nPlease insert path for recovering a snapshot or a completed model.\
- This can be a local directory or a GCS Bucket:\n")
+recovery_path = raw_input("\nInsert a path from which checkpoints will\
```
Comment (werkaaa) [inline user_interaction.py]: Same here

File: user_interaction.py [hunk 3]
```diff
@@ -1,110 +1,108 @@
...
-dataPath = raw_input("Please insert a path for data retrieving. This can be a local directory or a GCS Bucket:\n")
+# # Determining path for data retrieval
+data_path = raw_input("Insert a path for data retrieving. \
+This can be a local directory or a GCS Bucket:\n")
 
 # Determining path for storing model, snapshots and summaries
-storingPath = raw_input("\nPlease insert a path for storing the model, snapshots and summaries.\
- This can be a local directory or a GCS Bucket:\n")
+storing_path = raw_input("\nInsert a path for saving the model, \
+snapshots and summaries. This can be a local directory or a GCS Bucket:\n")
```
Comment (werkaaa) [inline user_interaction.py]: With current setup, I think it can be only GCS Bucket. If model is saved inside the VM it gets destroyed when the job finishes.

File: user_interaction.py [hunk 4]
```diff
@@ -1,110 +1,108 @@
...
-        continue
-    else:
-        break
-
-print("You chose to " + mode + " " + placeOptions[place] + "\n")
-
-# Determining path for data retrieval
-dataPath = raw_input("Please insert a path for data retrieving. This can be a local directory or a GCS Bucket:\n")
+# # Determining path for data retrieval
+data_path = raw_input("Insert a path for data retrieving. \
```
Comment (werkaaa) [inline user_interaction.py]: Dockerfile for Magenta library will only have the option of putting the data inside GCS bucket.

File: user_interaction.py [hunk 5]
```diff
@@ -1,110 +1,108 @@
...
-# Determining path for recovering a snapshot
-recoveryPath = raw_input("\nPlease insert path for recovering a snapshot or a completed model.\
- This can be a local directory or a GCS Bucket:\n")
+recovery_path = raw_input("\nInsert a path from which checkpoints will\
+ be restored before training. Skip for using the same path as for saving:\n")
+if recovery_path == "":
+    recovery_path = storing_path
 
-imageURI = raw_input("\nPlease insert a IMAGE URI:")
+gin_path = raw_input("\nInsert an additional gin file search path or skip\
```
Comment (werkaaa) [inline user_interaction.py]: As Jesse recommended me in my first PR, I think it is better to format the strings in this way:
('Some very '
'long string.')
instead of using \ to brake line.

File: user_interaction.py [hunk 6]
```diff
@@ -1,110 +1,108 @@
...
++ " --gin_file=models/solo_instrument.gin"\
++ " --gin_file=datasets/tfrecord.gin"\
++ " --gin_param=\"TFRecordProvider.file_pattern='" + data_path\
++ "/train.tfrecord*'\""\
++ " --gin_param=checkpoints_to_keep=" + no_of_checkpoints\
++ " --gin_param=early_stop_loss_value=" + early_stop_loss_value\
++ " --gin_param=magenta_ddsp_internals.train_util.train.batch_size=" + batch_size\
++ " --gin_param=magenta_ddsp_internals.train_util.train.num_steps=" + no_of_steps\
++ " --gin_param=magenta_ddsp_internals.train_util.train.steps_per_summary=" + steps_per_summary\
++ " --gin_param=magenta_ddsp_internals.train_util.train.steps_per_save=" + steps_per_save
```
Comment (werkaaa) [inline user_interaction.py]: I will add this super long flags support in the task.py script so the user will be able to use just short version, but I will keep you updated on it.

File: user_interaction.py [hunk 7]
```diff
@@ -1,110 +1,105 @@
...
+
+early_stop_loss_value = raw_input('\nInsert the early stop loss value'
+' or skip for the default value (5):')
+if early_stop_loss_value == "":
+    early_stop_loss_value = "5"
+
+# Submit the job on AI Platform
+print("Submitting the job on AI Platform")
+submitting_job = "gcloud beta ai-platform jobs submit training " + job_name\
++ " --region " + region + " --master-image-uri " + image_URI + " --config " + config_path + " -- --mode=train"\
```
Comment (werkaaa) [inline user_interaction.py]: Train mode will be automatically called in task.py

File: user_interaction.py [hunk 8]
```diff
@@ -1,110 +1,105 @@
...
++ " --save_dir=" + save_dir\
++ " --restore_dir=" + restore_dir\
++ " --gin_search_path=" + gin_path\
++ " --alsologtostderr"\
```
Comment (werkaaa) [inline user_interaction.py]: Logs will be also called inside task.py

File: user_interaction.py [hunk 9]
```diff
@@ -1,110 +1,105 @@
...
+if early_stop_loss_value == "":
+    early_stop_loss_value = "5"
+
+# Submit the job on AI Platform
+print("Submitting the job on AI Platform")
+submitting_job = "gcloud beta ai-platform jobs submit training " + job_name\
++ " --region " + region + " --master-image-uri " + image_URI + " --config " + config_path + " -- --mode=train"\
++ " --save_dir=" + save_dir\
++ " --restore_dir=" + restore_dir\
++ " --gin_search_path=" + gin_path\
```
Comment (werkaaa) [inline user_interaction.py]: This can be only local path inside the container. We would have to assume that user built the image with gin configs inside it. I guess it can be done this way but we have to write about it in the documentation.

File: user_interaction.py [hunk 10]
```diff
@@ -1,110 +1,105 @@
...
++ " --alsologtostderr"\
++ " --hp_tuning=True"\
++ " --gin_file=models/solo_instrument.gin"\
++ " --gin_file=datasets/tfrecord.gin"\
```
Comment (werkaaa) [inline user_interaction.py]: As those two gin_files define the whole structure of model and data I think it is better to also call it in task.py

File: Dockerfile [hunk 11]
```diff
@@ -1,60 +1,24 @@
+
```
Comment (marmolejo) [inline Dockerfile:1]: You can remove this blank line.

File: Dockerfile [hunk 12]
```diff
@@ -1,60 +1,24 @@
+ 
 FROM gcr.io/deeplearning-platform-release/tf2-gpu.2-2
 WORKDIR /root
```
Comment (marmolejo) [inline Dockerfile:3]: Move this workdir directive before line 16, that is when is actually needed.

File: Dockerfile [hunk 13]
```diff
@@ -1,60 +1,24 @@
...
 
 # Installs sndfile library for reading and writing audio files 
 RUN apt-get update && apt-get install -y libsndfile-dev
```
Comment (marmolejo) [inline Dockerfile:6]: add --no-installs-recommends to the apt-get command so it doesn't install extra unwanted packages.

File: Dockerfile [hunk 14]
```diff
@@ -1,60 +1,24 @@
...
+# Install cloudml-hypertune package needed for hyperparameter tuning
+RUN pip install cloudml-hypertune
 
-# Path configuration
-ENV PATH $PATH:/root/tools/google-cloud-sdk/bin
-# Makes sure gsutil will use the default service account
-RUN echo '[GoogleCompute]\nservice_account = default' > /etc/boto.cfg
+RUN wget https://github.com/ana-simionescu/ddsp/archive/master.zip
+RUN unzip master.zip
+RUN cd ddsp-master && python setup.py install
```
Comment (marmolejo) [inline Dockerfile:18]: To avoid checking in temporary files in the fs, merge these 3 commands in a single one:
```
RUN wget https://github.com/ana-simionescu/ddsp/archive/master.zip && \
  unzip master.zip && \
  cd ddsp-master && python setup.py install

File: Dockerfile [hunk 15]
```diff
@@ -1,60 +1,24 @@
...
 
 # These parameters can be also specified as part of the job submission arguments
-ENTRYPOINT ["python", "trainer/ddsp_run_multiple_vms.py", "--mode=train", \
-     "--alsologtostderr", \
-     "--gin_file=models/solo_instrument.gin", \
-     "--gin_file=datasets/tfrecord.gin"]
-
-
-
+ENTRYPOINT ["python", "task.py"]
```
Comment (marmolejo) [inline Dockerfile:24]: Add trailing newline

File: cloud.py [hunk 16]
```diff
@@ -0,0 +1,110 @@
+# Copyright 2020 The DDSP Authors.
```
Comment (marmolejo) [inline cloud.py:1]: These files part of DDSP, if you are importing them to this repository, better to add them in a different directory `third_party/ddsp` and make it clear they come from another source. However to see which are your contributions and preserve the history I would suggest doing the PR against ddsp repo directly.

File: magenta_docker/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (marmolejo) [inline magenta_docker/Dockerfile]: Remove blank line.

File: magenta_docker/Dockerfile [hunk 18]
```diff
@@ -0,0 +1,24 @@
+ 
+FROM gcr.io/deeplearning-platform-release/tf2-gpu.2-2
+WORKDIR /root
```
Comment (marmolejo) [inline magenta_docker/Dockerfile]: Move this WORKDIR directive just before line 16.

File: magenta_docker/Dockerfile [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+
+# Installs sndfile library for reading and writing audio files 
+RUN apt-get update && apt-get install -y libsndfile-dev
```
Comment (marmolejo) [inline magenta_docker/Dockerfile]: Add --no-install-recommends

File: magenta_docker/Dockerfile [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+# Newer version of Tensorflow is needed for multiple VMs training
+RUN pip install --upgrade pip && pip install --upgrade tensorflow tensorflow-probability
+RUN pip show tensorflow
+
+# Install cloudml-hypertune package needed for hyperparameter tuning
+RUN pip install cloudml-hypertune
+
+RUN wget https://github.com/ana-simionescu/ddsp/archive/master.zip
+RUN unzip master.zip
+RUN cd ddsp-master && python setup.py install
```
Comment (marmolejo) [inline magenta_docker/Dockerfile]: Merge these lines into one.

File: magenta_docker/user_interaction.py [hunk 21]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+import os
+
+# Determining path for data retrieval
+data_path = raw_input('Insert a path for data retrieving. '
+'This must be a GCS Bucket:\n')
```
Comment (marmolejo) [inline magenta_docker/user_interaction.py:6]: This line has to go under `'Insert...`. Have you ran this under pylint?

File: magenta_docker/user_interaction.py [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+
+# Determining path for storing model, snapshots and summaries
+save_dir = raw_input('\nInsert a path for saving the model, '
+'snapshots and summaries. This must be a GCS Bucket:\n')
```
Comment (marmolejo) [inline magenta_docker/user_interaction.py:10]: Same here.

File: magenta_docker/user_interaction.py [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+
+# Determining path for restoring the model
+restore_dir = raw_input('\nInsert a path from which checkpoints will '
+'be restored before training. Skip for using the same path as for saving:\n')
```
Comment (marmolejo) [inline magenta_docker/user_interaction.py:14]: Here and all occurrences below :)

File: magenta_docker/user_interaction.py [hunk 24]
```diff
@@ -0,0 +1,99 @@
...
+if steps_per_summary == "":
+    steps_per_summary = "300"
+
+early_stop_loss_value = raw_input('\nInsert the early stop loss value '
+'or skip for the default value (5): ')
+if early_stop_loss_value == "":
+    early_stop_loss_value = "5"
+
+# Submit the job on AI Platform
+print("Submitting the job on AI Platform")
```
Comment (marmolejo) [inline magenta_docker/user_interaction.py:76]: I would separate the input of paramenters with the processing. In this case, I would make 2 functions, one to get the inputs, the other to process.

File: user_interaction.py [hunk 25]
```diff
@@ -1,110 +1,98 @@
 #!/usr/bin/env python
```
Comment (marmolejo) [inline user_interaction.py:1]: What is the difference between this script and the one in magenta_docker directory?

**Review by marmolejo — CHANGES_REQUESTED** [review]
